docs/_includes/Expressions.md

```diff
@@ -418,21 +418,57 @@ or a collection of object references.
 
 TO BE WRITTEN -- allocated predicate
 
-## Old Expressions {#sec-old-expression}
+## Old and Old@ Expressions {#sec-old-expression}
 
 ````
-OldExpression_ = "old" [ "@" ident ] "(" Expression(allowLemma: true, allowLambda: true) ")"
+OldExpression_ = "old" "(" Expression(allowLemma: true, 
+                                      allowLambda: true) ")"
+               | "old" "@" label "(" Expression(allowLemma: true, 
+                                      allowLambda: true) ")"
 ````
 
-An _old expression_ is used in postconditions. `old(e)` evaluates to
-the value expression `e` had on entry to the current method.
-Note that **old** only affects heap dereferences, like `o.f` and `a[i]`.
-In particular, **old** has no effect on the value returned for local
-variables or out-parameters.
+An _old expression_ is used in postconditions or in the body of a method;
```

Comment:
or anywhere (spec or body) of a two-state function or two-state lemma

Reply:
OK